CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+# Contributing {#about_contributing}
+
+First of all, thanks for wanting to contribute to SeqAn! Community is important to us and we strive to maintain a great
+culture and atmosphere. Please have a look at our [Code of Conduct](\ref about_code_of_conduct).
+
+The following is a guide that helps stream-line the process of adding changes. If you haven't contributed to SeqAn
+before, don't worry about getting something wrong, that's absolutely OK. However, following this guide closely
+will reduce work for all of us and increases the chance of your changes being merged quickly.
+
+# Pull request workflow
+
+The overall workflow of contributing changes to this repository is:
+
+  1. Create an issue with bug report or feature request.
+  2. Wait for comments from the team and then collectively decide on the best way to solve the issue.
+  3. Fork the repository and create a branch on your fork.
+  4. Commit the solution to your local branch.
+  5. Create a pull-request to our master branch from your branch.
+  6. Wait for reviews and resolve reviewer comments.
+  7. Wait for the PR to be merged.
+
+## Creating an issue
+
+Go to https://github.com/seqan/seqan3/issues and check if someone else has already reported the same issue. If not,
+create a new issue and follow the template.
+
+We usually reply quickly to issues; if there is no reply within a week (and it's not prime vacation season in Europe)
+feel free to reply to the issue and explicitly mention `@seqan/core`.
+
+## Creating commits
+
+If you have never used the Fork & Pull Model, have a look at [this guide](https://guides.github.com/introduction/flow/).
+
+When creating commits to your branch and before requesting a pull, please ensure the following:
+
+  * Proper **naming** of commits (see below).
+  * Content other than the name is irrelevant in the commit message.
+  * The number of commits should be minimal and commits shouldn't revert/change previous ones.
+    * You can use separate commits for implementation/tests/documentation or for individual features/fixes.
+    * Do not add additional commits a la "Resolve reviewer comments", instead update old commits via `git rebase -i` or
```

Comment:
If we suggest rebasing, then we need to add a note or a link, that this might destroy the git tree of others if they have pulled the changes. I  know it is a rare case, but it should be at least made aware of.

Reply:
For a branch that someone created on their own fork to fix something in our repo?

Reply:
well, I have read guides to never rebase for the very same reason. Of course it is very unlikely but if we tell the people to do it I think they should at least know the implications. Something like please read the section to rebase on the git manual page, bla.